### PR TITLE
Yukon-Basic-Personal-Amount

### LINKED ISSUE #312: Yukon basic personal amount
reference:
https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1yt-ws/td1yt-ws-23e.pdf

### PR DESCRIPTION
**Pull request by sqw0418**
Fixes #312

### CODE REVIEW

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/amount.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+description: Yukon credits basic personal amount, based on the filers income.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/amount.yaml:1]:
```suggestion
description: Yukon credits this basic personal amount, based on the filers income.
```

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/amount.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      2022-01-01: 14_398
+      2023-01-01: 15_000
+  - threshold:
+      2022-01-01: 155_625
+      2023-01-01: 165_430
+    rate:
+      2023-01-01: 0.02105488 # 1_479 / 70_245
+  - threshold:
+      2022-01-01: 221_708 # 155_625 + 66_083
+      2023-01-01: 235_675 # 165_430 + 70_245
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/amount.yaml:15]: check logic - check form and legal code

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/amount.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+brackets:
+  - threshold:
+      2022-01-01: 0
+    rate:
+      2022-01-01: 14_398
+      2023-01-01: 15_000
+  - threshold:
+      2022-01-01: 155_625
+      2023-01-01: 165_430
+    rate:
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/amount.yaml:11]: we want to split this up into the parameters of the form

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/amount.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+description: Yukon credits basic personal amount, based on the filers income.
+brackets:
+  - threshold:
+      2022-01-01: 0
+    rate:
+      2022-01-01: 14_398
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/amount.yaml:6]:
```suggestion
      2022-01-01: 0
```

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/amount.yaml (hunk 3 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/amount.yaml:11]: keep this at reduction with rates and make a max amount file

File: policyengine_canada/tests/gov/provinces/yt/tax/income/credits/yt_basic_personal_amount.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  
+- name: Eligible & Exceeding
+  period: 2023
+  input:
+    province_code: YT
+    individual_net_income: 300_000
+  output:
+    yt_basic_personal_amount: 13_521
+
+- name: Not head
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/yt/tax/income/credits/yt_basic_personal_amount.yaml:25]: dont need this test

File: policyengine_canada/variables/gov/provinces/yt/tax/income/yt_basic_personal_amount.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+            period
+        ).gov.provinces.yt.tax.income.credits.basic_personal_amount
+
+        applicable_amount = p.applicable_amount
+        base_amount = p.base_amount
+        division = p.division
+        income_threshold = p.income_threshold
+        individual_net_income = person("individual_net_income", period)
+
+        eligible = (division - (individual_net_income - income_threshold)) <= 0
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/yt/tax/income/yt_basic_personal_amount.py:22]:
```suggestion
        additional_amount_eligible = (division - (individual_net_income - income_threshold)) <= 0
```

File: policyengine_canada/variables/gov/provinces/yt/tax/income/yt_basic_personal_amount.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+        eligible_amount = division - (individual_net_income - income_threshold)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/yt/tax/income/yt_basic_personal_amount.py:23]:
```suggestion
        additional_amount = division - (individual_net_income - income_threshold)
```

File: policyengine_canada/variables/gov/provinces/yt/tax/income/yt_basic_personal_amount.py [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    defined_for = ProvinceCode.YT
+
+    def formula(person, period, parameters):
+        p = parameters(
+            period
+        ).gov.provinces.yt.tax.income.credits.basic_personal_amount
+
+        applicable_amount = p.applicable_amount
+        base_amount = p.base_amount
+        division = p.division
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/yt/tax/income/yt_basic_personal_amount.py:18]:
```suggestion
        divisor = p.division
```

File: policyengine_canada/variables/gov/provinces/yt/tax/income/yt_basic_personal_amount.py (hunk 6 above)
Comment (PavelMakarchuk) [reply policyengine_canada/variables/gov/provinces/yt/tax/income/yt_basic_personal_amount.py:22]: put additional amount on top of the eligibility

File: policyengine_canada/variables/gov/provinces/yt/tax/income/yt_basic_personal_amount.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+        income_threshold = p.income_threshold
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/yt/tax/income/yt_basic_personal_amount.py:19]:
```suggestion
```

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml [hunk 10]
```diff
@@ -0,0 +1,18 @@
+description: Yukon credits this basic personal amount, based on the base amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml:1]: rewrite the descriptions

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+values:
+  2022-01-01: 13_521
+metadata:
+  unit: currency-CAD
+  label: Yukon income basic personal base amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml:6]: rewrite labels

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+  reference:
+  - title: Worksheet for the Yukon 2023 Personal Tax Credits Return
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1yt-ws/td1yt-ws-23e.pdf#page=1
+  # Basic Personal Amount does not appear in the Yukon income tax act
+  - title: Income Tax Act
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml:11]: add section of income tax act

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+    href: https://laws.yukon.ca/cms/images/LEGISLATION/PRINCIPAL/2002/2002-0118/2002-0118.pdf
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml:12]: add page

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml (hunk 12 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml:11]: all yaml

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml (hunk 13 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml:12]: all yaml

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml (hunk 11 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml:6]: all yaml

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml (hunk 10 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml:1]: all yaml

File: policyengine_canada/tests/gov/provinces/yt/tax/income/credits/yt_basic_personal_amount.yaml [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+  input:
+    province_code: YT
+    individual_net_income: 200_000
+  output:
+    yt_basic_personal_amount: 14_272
+
+
+
+
+
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/yt/tax/income/credits/yt_basic_personal_amount.yaml:21]: delete extra lines

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/division.yaml [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+  reference:
+  - title: Worksheet for the Yukon 2023 Personal Tax Credits Return
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1yt-ws/td1yt-ws-23e.pdf#page=1
+  # Basic Personal Amount does not appear in the Yukon income tax act
+  - title: Income Tax Act
+    href: https://laws.yukon.ca/cms/images/LEGISLATION/PRINCIPAL/2002/2002-0118/2002-0118.pdf
+  - title: 2023 Yukon Personal Tax Credits Return
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1yt/td1yt-23e.pdf#page=1
+  - title: 2022 Yukon Personal Tax Credits Return
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1yt/td1yt-22e.pdf#page=1
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/division.yaml:16]: delete all sources which do not directly reference the number

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/division.yaml:16]: all yaml files

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/division.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/division.yaml:1]: rename to divisor.yaml

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/applicable_amount.yaml [hunk 16]
```diff
@@ -0,0 +1,9 @@
+description: Yukon credits this basic personal amount, based on this applicable amount
+values:
+  2022-01-01: 1_479
+metadata:
+  unit: currency-CAD
+  label: Yukon income applicable amount
+  reference:
+  - title: Worksheet for the Yukon 2023 Personal Tax Credits Return
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/applicable_amount.yaml:8]: add 2022 reference and double check values

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/applicable_amount.yaml [hunk 17]
```diff
@@ -0,0 +1,9 @@
+description: Yukon credits this basic personal amount, based on this applicable amount
+values:
+  2022-01-01: 1_479
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/applicable_amount.yaml:3]:
```suggestion
  2022-01-01: 1_679
  2023-01-01: 1_479
```

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/applicable_amount.yaml (hunk 16 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/applicable_amount.yaml:8]: leave the law reference and add a comment: # The credit amount is not mentioned in the Yukon income tax act

https://laws.yukon.ca/cms/images/LEGISLATION/PRINCIPAL/2002/2002-0118/2002-0118.pdf

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/applicable_amount.yaml:8]: for every yaml file in this PR

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml [hunk 18]
```diff
@@ -0,0 +1,9 @@
+description: Yukon credits this basic personal amount, based on this base amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/base_amount.yaml:1]:
```suggestion
description: Yukon credits this base basic personal amount.
```

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/divisor.yaml [hunk 19]
```diff
@@ -0,0 +1,9 @@
+description: Yukon credits basic personal amount, this is divisor for the calculation
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/divisor.yaml:1]:
```suggestion
description: Yukon divides the basic personal amount by this amount.
```

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/divisor.yaml [hunk 20]
```diff
@@ -0,0 +1,9 @@
...
+values:
+  2022-01-01: 70_245
+metadata:
+  unit: currency-CAD
+  label: Yukon income basic personal divisor amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/divisor.yaml:6]:
```suggestion
  label: Yukon income basic personal divisor
```

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/income_threshold.yaml [hunk 21]
```diff
@@ -0,0 +1,9 @@
+description: Yukon credits this basic personal amount, based on the income threshold
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/income_threshold.yaml:1]:
```suggestion
description: Yukon reduces the basic personal amount for filers with income above this threshold.
```

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/income_threshold.yaml [hunk 22]
```diff
@@ -0,0 +1,9 @@
...
+values:
+  2022-01-01: 165_430
+metadata:
+  unit: currency-CAD
+  label: Yukon income threshold of basic personal amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/income_threshold.yaml:6]:
```suggestion
  label: Yukon basic personal amount reduction income threshold
```

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/applicable_amount.yaml [hunk 23]
```diff
@@ -0,0 +1,9 @@
+description: Yukon credits this basic personal amount, based on this applicable amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/applicable_amount.yaml:1]:
```suggestion
description: Yukon reduces this basic personal amount, for high income filers.
```

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/applicable_amount.yaml [hunk 24]
```diff
@@ -0,0 +1,9 @@
...
+values:
+  2022-01-01: 1_479
+metadata:
+  unit: currency-CAD
+  label: Yukon income applicable amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/applicable_amount.yaml:6]:
```suggestion
  label: Yukon basic personal amount applicable amount
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/applicable_amount.yaml [hunk 25]
```diff
@@ -0,0 +1,15 @@
+description: Yukon reduces this basic personal amount, for high income filers.
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/applicable_amount.yaml:1]: Isn't this the amount of the additional bpa?

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/divisor.yaml
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/basic_personal_amount/divisor.yaml:1]: this is the phase-out width

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]
Max left comments, I dont see any commits resolving those

**Pull request closed without merging**